### ISSUE TITLE
Will this be for version 8 pixi as of v8.0.0-beta.6 it is not working

### ISSUE THREAD

**Issue by xtianus79** [pixijs/graphics-smooth#44]
Hi will this work for version 8?

**Comment by ivanpopelyshev** [comment]
Its just tehre's no one who has time to do it :) 

@bigtimebuddy @GoodBoyDigital any way to make a bounty on it from opencollective?

**Comment by xtianus79** [comment]
hi @ivanpopelyshev @bigtimebuddy  and @GoodBoyDigital  I am willing to pay a bounty if you guys will help. this is vital to my project.

**Comment by gnut-neygun** [comment]
I'm also highly interested in this. As the transform-independent line width feature from this package is vital to my project.

**Comment by bigtimebuddy** [comment]
@xtianus79 if you are interested in sponsoring this plugin, please see [this](https://github.com/pixijs/pixijs/wiki/Bounties#--sponsoring). Happy to facilitate if we can find a hero to do the work.

**Comment by gnut-neygun** [comment]
I am actually looking at this to see what I can do myself. But I believe any PRs can not proceed until those @pixi packages for version 8 are published first on npm. (currently there's only pixi.js@pre-release)

**Comment by bigtimebuddy** [comment]
`@pixi` packages are not being shipped for v8 and are being deprecated. Only `pixi.js` will be shipped for v8

**Comment by MuTsunTsai** [comment]
I was really looking forward to this, but it then seems that the built-in Graphics of the latest Pixi v8 is already good enough if we have something like this:

```js
const app = new Application();
await app.init({
    // A combination of the following three gives pretty good rendering
    antialias: true,
    resolution: 2,
    autoDensity: true,
    ...
});
```

I made an [example](https://mutsuntsai.github.io/pixi-v8-test/) mimicking the "Joint" example. It seems to me that the result is comparably good to Pixi v7 + graphics-smooth. So do we still need to work on v8 support? What do you guys think?

**Comment by ivanpopelyshev** [comment]
`antialias:true` + `resolution:2` is a performance sink, that's why graphics-smooth exists. For pixi-v8 we need custom graphics batchers first, I'm waiting on this thing: https://github.com/pixijs/pixijs/pull/10702

**Comment by Specy** [comment]
Is there some update about this issue? I'm using the library to dynamically create sprites that I then use as textures and it's blocking me from updating to V8. If this won't be worked on, is there an alternative I could use to keep the smooth/antialias look on the textures?

**Comment by MuTsunTsai** [comment]
@ivanpopelyshev So accordingly Pixi now supports custom batchers (since [v8.4](https://github.com/pixijs/pixijs/releases/tag/v8.4.0) actually) and pixijs/pixijs#10702 has been closed. Hopefully that means we will see v8 support soon. I'm more than willing if there's anything I can help, as this has been blocking my project from updating Pixi for so long.

**Comment by ivanpopelyshev** [comment]
> @ivanpopelyshev So accordingly Pixi now supports custom batchers (since [v8.4](https://github.com/pixijs/pixijs/releases/tag/v8.4.0) actually) and [pixijs/pixijs#10702](https://github.com/pixijs/pixijs/pull/10702) has been closed. Hopefully that means we will see v8 support soon. I'm more than willing if there's anything I can help, as this has been blocking my project from updating Pixi for so long.

Yes

**Comment by GeorgeBarlow** [comment]
Is there any update regarding the v8 progress, this is a big plugin for us and we'd love to see it in v8 :)

**Comment by plkoiu1122** [comment]
Any update in this.

**Comment by ivanpopelyshev** [comment]
> Any update in this.

This month probably. Its .. several months like that )

**Comment by leapAbhishek** [comment]
Any update guys, its been over a year. Been waiting for v8 support so bad.

**Comment by ivanpopelyshev** [comment]
Update: im still too busy on my main job :( If you know someone else who can do it- he can adopt this repo

**Comment by GeorgeBarlow** [comment]
> Update: im still too busy on my main job :( If you know someone else who can do it- he can adopt this repo

No problem! If you get some time, would you be able to brainstorm what challenges (if any) you were anticipating with this conversion to v8? Also, if possible, any key focus points/areas us individuals would need to be aware of if we were to look into trying to help with the transition to v8?

It seems like a lot of people really want this, and it would be great to try and get the ball rolling - I'm happy to help but I personally have only worked with Pixi for the last 6 months or so, so it would be useful to have some general pointers into what needs to be done.  :)

**Comment by ivanpopelyshev** [comment]
Easy, shader has to be converted to TS format for my own shader compiler that produces both GLSL and WGSL, and API converted to v8. Also Graphics stuff is different in v7 and v8, and not in a good sense, we might have more problems with new line styles and stuff

**Comment by ivanpopelyshev** [comment]
its just too much, and to debug EVERYTHING at the same time might be difficult. There wer many "black screen" problems last time i did that. Also, shader is ~150 lines of code , and maintaing separate versions for glsl wgsl just not possible, need to use either threejs TSL either my own thing that unfortunately cant be open-sourced yet.

**Comment by cangerjun** [comment]
1.5 years later........
We haven't succeeded yet......

**Comment by Specy** [comment]
@cangerjun this is open source software, none has a responsibility or obligation to work on it, especially since most of the time it's work done for free. It will be worked on whenever someone has the motivation or a reason to do so (same thing applies to the issue on my repository which links to this issue)

**Comment by ivanpopelyshev** [comment]
> 1.5 years later........ We haven't succeeded yet......

yeah, kinda my fault ) hope to try fix it again in september

**Comment by ivanpopelyshev** [comment]
The problem is - this thing is eating almost all my time: https://tesera.io , i dont even have enough power to make PR's for everything we discovered in pixi8

**Comment by msurguy** [comment]
Can AI do this upgrade? willing to pitch in some $$$ for AI credits... If you were to write out a prompt for AI (context engineering), what would it be?

**Comment by ivanpopelyshev** [comment]
> Can AI do this upgrade? willing to pitch in some $$$ for AI credits... If you were to write out a prompt for AI (context engineering), what would it be?

Write me in pixijs discord (Hackerham), if you want to sponsor this. no AI, I'm free for this week, can do this thing.

**Comment by Haicursor** [comment]
Any update in this?

**Comment by GeorgeBarlow** [comment]
@ivanpopelyshev did the sponsor stuff ever come through? still waiting patiently haha

**Comment by Bram1903** [comment]
This project looks really interesting. Would love to use this on V8!

**Comment by elHornair** [comment]
I'm currently working on a Pixi V8 project where it's crucial to have polygons with dashed pixelLine edges. There is budget to sponsor this. @bigtimebuddy @ivanpopelyshev would you be willing to do this? How could we proceed?

Details / Background:
As a POC, I'm drawing them as a sequence of moveTo / lineTo statements (as suggested by @bigtimebuddy elsewhere I believe). This works but is a performance killer with > 1 million points involved. Based on [this conversation](https://github.com/pixijs/pixijs/discussions/11830) and [this code](https://github.com/pixijs-userland/graphics-smooth/blob/main/src/DashLineShader.ts) it seems that a more performant way of doing this is with a custom VertexShader. My gut tells me that this is not a trivial thing to do (mostly because the dashes don't just perfectly stop at a vertex but continue going on around it; but also just in general). If anyone managed to do this already and could share some insights, I would greatly appreciate it :)

**Comment by elHornair** [comment]
I made this little POC for dashed pixelLines: https://stackblitz.com/edit/vitejs-vite-tmtukpm6?file=src%2Fmain.ts. Maybe this is useful to someone. Any feedback or suggestions for improvement is welcome too.

**Comment by nebsar** [comment]
Yeah I also used Mesh for drawing a dashed line. But the batching does not work with Mesh and I think there is no mouse interaction capability with Mesh.

**Comment by NIKENIT** [comment]
So, any news on this?

**Comment by bigtimebuddy** [comment]
We have better AI tools in 2026 than 2023, someone want to take an attempt at this?

**Comment by MuTsunTsai** [comment]
@bigtimebuddy Yes, definitely. But I'll need something to start with. A work in progress, or some basic examples to extend from, perhaps?

**Comment by ivanpopelyshev** [comment]
I tried to merge smooth with v8 graphics and what can I say: v8 graphics is made completely incompatible with v7 code, I do not understand what the heck people who were implementing it thought. You probably need access to Mythos to marry v7 and v8.

I can port v7 code that wont be compatible with v8 graphics, just using same API's but all internals will be different. Alternatively, I can write better Graphics, with all extra features that I know people want.

**Comment by ivanpopelyshev** [comment]
Yeah, I didnt notice it: https://github.com/pixijs/pixijs/pull/12002 , its being implemented in main pixi repo, with support of custom Graphics shaders too. I was too by my fail at trying to hack it from plugin scope, my bad.

Gonna test it!

**Comment by ivanpopelyshev** [comment]
Its close to corrent but chipped, gonna check whats wrong with input to shaders there

**Comment by ivanpopelyshev** [comment]
I'm sorry for earlier not-so-constructive outburst. 
I made changes to PR that its almost ready for main pixi and has only a few checks left.
